src/cmd-cloud-prune

```diff
@@ -0,0 +1,300 @@
+    if m is None:
+        raise Exception(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return timestamp
+
+
+def get_json_from_s3(s3, bucket, key):
+    try:
+        response = s3.get_object(Bucket=bucket, Key=key)
+        content = response["Body"].read().decode("utf-8")
+        return json.loads(content)
+    except Exception as e:
+        raise Exception(f"Error fetching the JSON file from S3 {bucket}/{key}: {e}")
+
+
+def save_builds_json(builds_json_data):
+    with open(BUILDFILES['list'], "w") as json_file:
+        json.dump(builds_json_data, json_file, indent=2)
+
+
+def handle_upload_builds_json(s3_client, bucket, prefix, dry_run, acl):
+    remote_builds_json_data = get_json_from_s3(s3_client, bucket, os.path.join(prefix, "builds.json"))
+    with open(BUILDFILES['sourcedata'], "r") as f:
+        builds_json_source_data = json.load(f)
+    # Check if there are any changes that were made to remote(s3 version) builds.json
+    # while the pruning was in progress
+    if remote_builds_json_data != builds_json_source_data:
+        print("Detected remote updates to builds.json. Merging it to the local builds.json file")
+        with open(BUILDFILES['list'], "r") as f:
+            current_builds_json = json.load(f)
+        merged_builds_json = current_builds_json | remote_builds_json_data
+        if not dry_run:
+            # Make sure we have the merged json as local builds/builds.json
+            save_builds_json(merged_builds_json)
+        # Upload the local builds.json to s3
+        s3_copy(s3_client, BUILDFILES['list'], bucket, f'{prefix}/builds.json', CACHE_MAX_AGE_METADATA, acl, extra_args={}, dry_run=dry_run)
+    else:
+        s3_copy(s3_client, BUILDFILES['list'], bucket, f'{prefix}/builds.json', CACHE_MAX_AGE_METADATA, acl, extra_args={}, dry_run=dry_run)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(prog="coreos-assembler cloud-prune")
+    parser.add_argument("--policy", required=True, type=str, help="Path to policy YAML file")
+    parser.add_argument("--dry-run", help="Don't actually delete anything", action='store_true')
+    parser.add_argument("--upload-builds-json", help="Push builds.json", action='store_true')
+    parser.add_argument("--stream", type=str, help="CoreOS stream", required=True)
+    parser.add_argument("--gcp-json-key", help="GCP Service Account JSON Auth", default=os.environ.get("GCP_JSON_AUTH"))
+    parser.add_argument("--gcp-project", help="GCP Project name", default=os.environ.get("GCP_PROJECT_NAME"))
+    parser.add_argument("--acl", help="ACL for objects", action='store', default='private')
+    parser.add_argument("--aws-config-file", default=os.environ.get("AWS_CONFIG_FILE"), help="Path to AWS config file")
+    return parser.parse_args()
```

Comment:
Can we move that towards the top ? When reading a script it's easier to have an idea of how it's meant to be used first

Reply:
Agree with this, but I wouldn't consider it a blocker.